contracts/governance/ProposalValidator.sol

```diff
@@ -36,123 +42,120 @@ contract ProposalValidator is IProposalValidator {
+   * @param endTime timestamp when vote ends
+   * @param options list of proposal vote options
+   * @param daoOperator address of daoOperator
+   * @return boolean, true if can be created
    **/
-  function validateProposalCancellation(
-    IKyberGovernance governance,
-    address user,
-    uint256 blockNumber
-  ) external override view returns (bool) {
-    return !isPropositionPowerEnough(governance, user, blockNumber);
+  function validateGenericProposalCreation(
+    IVotingPowerStrategy strategy,
+    address creator,
+    uint256 startTime,
+    uint256 endTime,
+    string[] calldata options,
+    address daoOperator
+  ) external view returns (bool) {
+    // check authorization
+    if (creator != daoOperator || !isPropositionPowerEnough(strategy, creator, startTime)) return false;
+    // check vote duration
+    if (endTime.sub(startTime) < MIN_VOTING_DURATION) return false;
+    // check options length
+    if (options.length > MAX_VOTING_OPTIONS) return false;
+
+    return strategy.validateProposalCreation(creator, startTime, endTime);
   }
 
   /**
-   * @dev Returns whether a user has enough Proposition Power to make a proposal.
-   * @param governance Governance Contract
-   * @param user Address of the user to be challenged.
-   * @param blockNumber Block Number against which to make the challenge.
-   * @return true if user has enough power
+   * @dev Called to validate the cancellation of a proposal
+   * @param governance governance contract to fetch proposals from
+   * @param proposalId Id of the generic proposal
+   * @param user entity initiating the cancellation
+   * @param daoOperator address of daoOperator
+   * @return boolean, true if can be cancelled
    **/
-  function isPropositionPowerEnough(
+  function validateProposalCancellation(
     IKyberGovernance governance,
+    uint256 proposalId,
     address user,
-    uint256 blockNumber
-  ) public override view returns (bool) {
-    IGovernanceStrategy currentGovernanceStrategy = IGovernanceStrategy(
-      governance.getGovernanceStrategy()
-    );
-    return
-      currentGovernanceStrategy.getPropositionPowerAt(user, blockNumber) >=
-      getMinimumPropositionPowerNeeded(governance, blockNumber);
-  }
-
-  /**
-   * @dev Returns the minimum Proposition Power needed to create a proposition.
-   * @param governance Governance Contract
-   * @param blockNumber Blocknumber at which to evaluate
-   * @return minimum Proposition Power needed
-   **/
-  function getMinimumPropositionPowerNeeded(IKyberGovernance governance, uint256 blockNumber)
-    public
-    override
-    view
-    returns (uint256)
-  {
-    IGovernanceStrategy currentGovernanceStrategy = IGovernanceStrategy(
-      governance.getGovernanceStrategy()
+    address daoOperator
+  ) external view returns (bool) {
+    IKyberGovernance.ProposalWithoutVote memory proposal = governance.getProposalById(proposalId);
+    return (
+      user == daoOperator ||
```

Comment:
we already have checked user == daoOperator in the governance.

Reply:
ah... remove from gov or proposalvalidator?

Reply:
can remove from Proposal Validator, need to keep it in Governance to make sure that daoOperator can always cancel a proposal